src/utils/explorer-api.js

```diff
@@ -54,3 +54,32 @@ export async function getTransactions(accountId) {
 }
 
 export const transactionExtraInfo = (hash, signer_id) => wallet.connection.provider.sendJsonRpc('tx', [hash, signer_id])
+
+export async function getAccountId(publicKey) {
+    if (!publicKey) return {}
+
+    const accountId = await new Promise((resolve, reject) => wamp.call(
+        `${WAMP_NEAR_EXPLORER_TOPIC_PREFIX}.select`,
+        [
+            `
+                SELECT
+                    account_id
+                FROM 
+                    access_keys
+                WHERE 
+                    public_key = :publicKey
+            `,
+            { publicKey }
+        ],
+        {
+            onSuccess(dataArr) {
+                resolve(dataArr[0][0] && dataArr[0][0].account_id)
+            },
+            onError(err) {
+                reject(err);
+            }
+        }
+    ));
+
+    return accountId
```

Comment:
@marcinbodnar there can be multiple matching accounts. We'll need to recover all of them, but only select one. I guess selecting most recent one makes sense if we can order by creation time.

Reply:
It seems like `access_keys` table stores  `access_key_type`, `account_id` and `public_key` only. At this moment we are taking the first one:
https://github.com/near/near-wallet/pull/674/files#diff-8240df5baf1b5b5a5233c10523e59958R76
, but I can change it to the last one (more likely the last one will be the most recent one)?